src/python/espressomd/interactions.pyx

```diff
@@ -487,18 +479,7 @@ IF LENNARD_JONES_GENERIC == 1:
             """Python dictionary of default parameters.
 
             """
-            return {
-                "epsilon": 0.,
-                "sigma": 0.,
-                "cutoff": 0.,
-                "shift": 0.,
-                "offset": 0.,
-                "e1": 0,
-                "e2": 0,
-                "b1": 0.,
-                "b2": 0.,
-                "delta": 0.,
-                "lam": 0.}
+            return {"delta": 0., "lam": 0.}
```

Comment:
why are only those two parameters default ones? Shouldn't we reflect the facts from the core here?

Reply:
* `lam` is now 1 by default to reflect the core
* this line is now guarded by `LJGEN_SOFTCORE`
* keeping them as optional allows this generic LJ class to behave the same way when `LJGEN_SOFTCORE` is enabled, unless the user provides custom values for `lam` and `delta`